compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala

```diff
@@ -348,9 +354,11 @@ object TypeTestsCasts {
         if (sym.isTypeTest) {
           val argType = tree.args.head.tpe
           val isTrusted = tree.hasAttachment(PatternMatcher.TrustedTypeTestKey)
-          if (!isTrusted && !checkable(expr.tpe, argType, tree.span))
-            report.uncheckedWarning(i"the type test for $argType cannot be checked at runtime", expr.srcPos)
-          transformTypeTest(expr, tree.args.head.tpe,
+          val unchecked = expr.tpe.widenTermRefExpr.hasAnnotation(defn.UncheckedAnnot)
```

Comment:
No need to encode isTrusted as empty strings.

I think renaming `checkable` to `whyUncheckable` and changing the logic as below would make things clearer.
```scala
val isUnchecked = expr.tpe.widenTermRefExpr.hasAnnotation(defn.UncheckedAnnot)
if !isTrusted && !isUnchecked then
  val whyNot = whyUncheckable(expr.tpe, argType, tree.span)
  if !whyNot.isEmpty then 
    report.uncheckedWarning
```